### ISSUE TITLE
[bug] didnt find Cargo.toml

### ISSUE THREAD

**Issue by alexzanderr** [rust-windowing/android-ndk-rs#250]
i have this project

![image](https://user-images.githubusercontent.com/54772464/159593769-e226e15f-dffa-41fd-a06b-3142c9fc5a4b.png)

lib.rs
```rust
#[cfg_attr(target_os = "android", ndk_glue::main(backtrace = "on"))]
pub fn main() {
    println!("hello world");
}
```

main.rs
```rust
fn main() {
    $crate::main();
}
```

i have installed `cargo-apk` with `cargo install cargo-apk`

when im trying to run it with:
```shell
cargo apk run
```

it says `Error: Didn't find Cargo.toml.`

ummm. what im i doing wrong ?

Cargo.toml is definitely right there!

and btw, how do i install `ndk` (`#[cfg_attr(target_os = "android", ndk_glue::main(backtrace = "on"))]`) or `its built-in` ?

**Comment by MarijnS95** [comment]
`cargo-apk` can indeed be _much_ more helpful in its error codes, and this particular issue has recently been addressed (for context: https://github.com/dvc94ch/cargo-subcommand/pull/14/commits/dd765d1bbf978ace56b0bbddc8f6dbd8d775d5d0) and should reach end-users with the next release of `cargo-apk` which I'd like to get out the door very soon [^1].

[^1]: @dvc94ch perhaps we can do a `0.5.2` `cargo-subcommand` release on https://github.com/dvc94ch/cargo-subcommand/commit/7166d6f45a42105decb0244a61be0e746377269c so that it reaches everyone installing the current release of `cargo-apk`, pending the merge of #249 so that I can publish a new `cargo-apk` version.

Your project isn't a valid cargo crate setup. There's no `src/` directory and the `Cargo.toml` file is missing mandatory keys. Note that the guide in the readme should be read as a _migration_ on top of a crate that is already working, as such the `[lib] crate-type = ...` bit is _added_ to an existing and valid `Cargo.toml`. We should reword the readme to be more clear about this.

Try setting it up so that you can compile/run this hello-world application on your desktop first - a good start is `cargo new my_android_project --lib`. Bits like `#[cfg_attr(target_os = "android", ...)]` facilitate the use of your crate outside Android.

**Comment by MarijnS95** [comment]
Also note that you should _never_ name your project/crate `android`. That will end up in a `libandroid.so` binary which is the name of the framework library, and in turn fail to link.

Don't forget to run `cargo clean` if you've already built with this configuration, or a stray `libandroid.so` will linger in the `./target` folder.

https://github.com/rust-windowing/android-ndk-rs/search?q=libandroid.so&type=issues

**Comment by MarijnS95** [comment]
@alexzanderr check out #251: I've made some changes to the readme to make it easier to follow along. Please follow along and do comment on anything that is unclear!

**Comment by alexzanderr** [comment]
> `cargo-apk` can indeed be _much_ more helpful in its error codes, and this particular issue has recently been addressed (for context: https://github.com/dvc94ch/cargo-subcommand/pull/14/commits/dd765d1bbf978ace56b0bbddc8f6dbd8d775d5d0) and should reach end-users with the next release of `cargo-apk` which I'd like to get out the door very soon [^1].
> 
> [^1]: @dvc94ch perhaps we can do a `0.5.2` `cargo-subcommand` release on https://github.com/dvc94ch/cargo-subcommand/commit/7166d6f45a42105decb0244a61be0e746377269c so that it reaches everyone installing the current release of `cargo-apk`, pending the merge of #249 so that I can publish a new `cargo-apk` version.
> 
> Your project isn't a valid cargo crate setup. There's no `src/` directory and the `Cargo.toml` file is missing mandatory keys. Note that the guide in the readme should be read as a _migration_ on top of a crate that is already working, as such the `[lib] crate-type = ...` bit is _added_ to an existing and valid `Cargo.toml`. We should reword the readme to be more clear about this.
> 
> Try setting it up so that you can compile/run this hello-world application on your desktop first - a good start is `cargo new my_android_project --lib`. Bits like `#[cfg_attr(target_os = "android", ...)]` facilitate the use of your crate outside Android.

omg, i was so tired...

how could i make such a stupid mistake

i actually know that was invalid cargo project, thanks for response btw.

i ll give it a try and tell you what happened

and about ndk library, i guess i need to install ndk_glue only for quickstart right?

**Comment by MarijnS95** [comment]
> and about ndk library, i guess i need to install ndk_glue only for quickstart right?

Yes. You only need the `ndk` when directly interfacing with Android's NDK APIs, ie. what is available through https://docs.rs/ndk.

**Comment by alexzanderr** [comment]
so that's everything i need to build and run this android app made in rust?

by running the command cargo apk run will this popup an emulator on my Linux machine or popup to the mobile phone if connected?

i didnt see any gui library inside the startup code. how this app will show hello world inside a gui window?

am i missing something again?

**Comment by alexzanderr** [comment]
okey i've cloned the repo and tried to run the simplest example.
the one from `ndk-examples` with this command:
```shell
❱ [../android-ndk-rs/ndk-examples]
cargo apk run --example hello_world
```
and im getting this
```shell
❱ [../android-ndk-rs/ndk-examples]
Error: Android SDK is not found. Please set the path to the Android SDK with the $ANDROID_SDK_ROOT environment variable.
```

umm. how do i install android sdk or from where do i install android sdk ?
